File: src/services/roleService.js
import api from "../api/axios";

// Get All Roles
export const getRoles = () => {
  return api.get("/roles");
};

// Get Role By Id
export const getRoleById = (id) => {
  return api.get(`/roles/${id}`);
};

// Create Role
export const createRole = (role) => {
  return api.post("/roles", role);
};

// Update Role
export const updateRole = (id, role) => {
  return api.put(`/roles/${id}`, role);
};

// Delete Role
export const deleteRole = (id) => {
  return api.delete(`/roles/${id}`);
};

File: src/main/java/com/loan/service/RoleService.java
package com.loan.service;

import com.loan.dto.RoleRequest;
import com.loan.entity.Permission;
import com.loan.entity.Role;

import java.util.List;

public interface RoleService {

    List<Role> getAllRoles();

    Role getRoleById(Long id);

    Role createRole(RoleRequest request);

    Role updateRole(Long id, RoleRequest request);

    void deleteRole(Long id);

    /*
     * Returns permissions that the currently logged-in
     * user is allowed to assign to another role.
     *
     * ADMIN  -> all permissions
     * Others -> only their own role permissions
     */
    List<Permission> getAvailablePermissions();
}